packages/react-core/src/components/Banner/Banner.tsx

```diff
@@ -1,39 +1,68 @@
 import * as React from 'react';
 import styles from '@patternfly/react-styles/css/components/Banner/banner';
 import { css } from '@patternfly/react-styles';
+import { variantIcons } from '../Alert/AlertIcon';
+import { Flex, FlexItem } from '../../layouts';
 
 export interface BannerProps extends React.HTMLProps<HTMLDivElement> {
-  /** Content rendered inside the banner */
+  /** Content rendered inside the banner. */
   children?: React.ReactNode;
-  /** Additional classes added to the banner */
+  /** Additional classes added to the banner. */
   className?: string;
-  /** Variant styles for the banner */
-  variant?: 'default' | 'info' | 'danger' | 'success' | 'warning';
+  /** A custom icon for the banner. This property will override the icon that is set based on
+   * the variant property.
+   */
+  customIcon?: React.ReactNode;
+  /** Flag for indicating whether the banner has a status icon. When set to "true", the icon
+   * will be set based on the variant property.
+   */
+  hasStatusIcon?: boolean;
   /** If set to true, the banner sticks to the top of its container */
   isSticky?: boolean;
-  /** Text announced by screen readers to indicate the type of banner. Defaults to "${variant} banner" if this prop is not passed in */
+  /** Text announced by screen readers to indicate the type of banner when the hasStatusIcon property
+   * is passed in. Defaults to "${variant} banner" if this property is not passed in.
+   */
   screenReaderText?: string;
+  /** Variant styles for the banner. */
+  variant?: 'default' | 'info' | 'danger' | 'success' | 'warning';
 }
 
 export const Banner: React.FunctionComponent<BannerProps> = ({
   children,
   className,
+  customIcon,
+  hasStatusIcon = false,
   variant = 'default',
   screenReaderText,
   isSticky = false,
   ...props
-}: BannerProps) => (
-  <div
-    className={css(
-      styles.banner,
-      styles.modifiers[variant as 'success' | 'danger' | 'warning' | 'info'],
-      isSticky && styles.modifiers.sticky,
-      className
-    )}
-    {...props}
-  >
-    {children}
-    <span className="pf-u-screen-reader">{screenReaderText || `${variant} banner`}</span>
-  </div>
-);
+}: BannerProps) => {
+  const StatusIcon = variantIcons[variant];
+
+  return (
+    <div
+      className={css(
+        styles.banner,
+        styles.modifiers[variant as 'success' | 'danger' | 'warning' | 'info'],
```

Comment:
FWIW we changed the "status" variant names to color variants since the color banner used doesn't always convey status. AFAIK we plan on removing the status variants in favor of the color names in the major release.

Reply:
That's something I was considering when working on this and ended up opting to keep this as-is for PF4. If it would make sense to update the class that gets applied to the `pf-m-[color]` classes I can do that, though, since we should be able to allow consumers to pass in `danger` etc as the variant and have that just apply `pf-m-red` etc.